src/ansi-c/c_typecheck_expr.cpp

```diff
@@ -2109,6 +2111,65 @@ exprt c_typecheck_baset::do_special_functions(
 
     return same_object_expr;
   }
+  else if(identifier == CPROVER_PREFIX "points_to_valid_memory")
+  {
+    if(expr.arguments().size() != 2 && expr.arguments().size() != 1)
+    {
+      error().source_location = f_op.source_location();
+      error() << "points_to_valid_memory expects one or two operands" << eom;
+      throw 0;
+    }
+    if(!is_lvalue(expr.arguments().front()))
+    {
+      error().source_location = f_op.source_location();
+      error() << "ptr argument to points_to_valid_memory"
+              << " must be an lvalue" << eom;
+      throw 0;
+    }
+
+    exprt same_object_expr;
+    if(expr.arguments().size() == 2)
+    {
+// TODO: We should add some way to enforce that the second argument
+// has a reasonable type (coercible to __CPROVER_size_t).
```

Comment:
Any suggestions about this?

Reply:
How about testing whether it's an `ID_unsignedbv` or `ID_signedbv` - I think that should do the trick.